### ISSUE TITLE
Not able to replicate your samples

### ISSUE THREAD

**Issue by bwatkin79** [marcosan93/Price-Forecaster#2]
Is there a reason that when I run your code it gives me 100 epochs, but only 34 samples? Yours is showing you had 792 samples. I don't see what is causing me to have a fraction of the samples you had. The only things I changed were that I'm using the prices for GE and the dates line changed to:

`plt.xlim('2017-04',`` '2021-04')`

 My dataset has 1259 rows of data to match your for testing. When I run print(df.shape), it gives me (1259, 85).

Also, two questions - Is adding volume to the analysis really as easy as just removing "Volume" from the df.drop line? Seems too easy. Does df.tail(10) have a function or was that just to test what was in the dataframe after normalizing?

**Comment by bhaskatripathi** [comment]
I had changed these two methods : It worked for me after these changes
#updated split_sequence for choosing last column

def split_sequence(seq, n_steps_in, n_steps_out):
    """
    Splits the multivariate time sequence
    """
    # Creating a list for both variables
    X, y = [], []
    for i in range(len(seq)):
        # Finding the end of the current sequence
        end = i + n_steps_in
        out_end = end + n_steps_out
        # Breaking out of the loop if we have exceeded the dataset's length
        if out_end > len(seq):
            break
        # Splitting the sequences into: x = past prices and indicators, y = prices ahead
        seq_x, seq_y = seq[i:end, :], seq[end:out_end, **-1**] #For choosing last column
        X.append(seq_x)
        y.append(seq_y)
    return np.array(X), np.array(y)

**#Updated the first columns[0] to columns[len(df.columns)-1]**
def validater(n_per_in, n_per_out):
    """
    Runs a 'For' loop to iterate through the length of the DF and create predicted values for every stated interval
    Returns a DF containing the predicted values for the model with the corresponding index values based on a business day frequency
    """
    # Creating an empty DF to store the predictions
    predictions = pd.DataFrame(index=df.index, columns=[df.columns[len(df.columns)-1]])
    for i in range(1, len(df)-n_per_in, n_per_out):
        # Creating rolling intervals to predict off of
        x = df[-i - n_per_in:-i]
        # Predicting using rolling intervals
        yhat = model.predict(np.array(x).reshape(1, n_per_in, n_features))
        # Transforming values back to their normal prices
        yhat = close_scaler.inverse_transform(yhat)[0]
        # DF to store the values and append later, frequency uses business days
        #plus_one_hour = datetime.datetime.strptime("%H:%M:%S") + datetime.timedelta(hours=2)
        start_dt= dt.datetime.strptime(x.index[-1], '%m/%d/%Y')+timedelta(days=1)
        dt_range=pd.date_range(start=start_dt, periods=len(yhat), freq="B")
        pred_df = pd.DataFrame(yhat,index=dt_range,columns=[x.columns[len(df.columns)-1]])
        # Updating the predictions DF
        print("Updating the prediction=",i)
        predictions.update(pred_df)
    return predictions

**Comment by bhaskatripathi** [comment]
However, he has not split the data to X-train,X-test, y-train,y-test so the model will be always biased due to peek ahead

**Comment by bwatkin79** [comment]
A couple questions for you on your posts. On your previous post, where did the dt come from? See below:
`start_dt= dt.datetime.strptime(x.index[-1], '%m/%d/%Y')+timedelta(days=1)`
 It was not in the original code, so I wasn't sure where it was added to make your code work. 

As to the quote below, how would you overcome the peek ahead issue? I've had one class that covered some machine learning, so this is very new to me. One thing I added to the program was Early Stopping to help prevent overfitting. Next thing I want to add is to retain the weights because it doesn't mean much if you find the right weighting, then can't reproduce it. 

To address your concern, I have something that comes to mind, but I doubt it's the most effective solution. Just throwing my inexperience out there. I'm picturing retaining the weights like I feel is already necessary. Weights would be tested in x and y test. Not sure how early stopping would work then because it's stopping based on the epoch results. What would you do?  Where would you build it in?

> However, he has not split the data to X-train,X-test, y-train,y-test so the model will be always biased due to peek ahead
